storage/remote/write.go

```diff
@@ -154,6 +126,16 @@ func (rws *WriteStorage) ApplyConfig(conf *config.Config) error {
 		if err != nil {
 			return err
 		}
+
+		queue, ok := rws.queues[hash]
+		if externalLabelUnchanged && ok {
+			// Update the client in case any secret configuration has changed.
```

Comment:
I'm not understanding why we need this. None of the http client config fields are hidden/missing yaml tags, so shouldn't they all be included when hashing the configs?

Reply:
Secrets are always marshaled into the value “Secret” so if the value changes we will not know from the hash.

Reply:
Hmm, yeah I'd missed that. Kind of inconvenient here for the purpose of generating a hash.

Reply:
I also think we should not be storing secrets in the hash, even if we use a cryptographic function. We could maybe change the whole thing to use DeepEqual, though reordering becomes messier then.

Reply:
It would just be nice to have one check for if anything in the config had changed :)